File: app/src/main/java/com/example/initialproject/presenter/PresenterImpl.java
package com.example.initialproject.presenter;

import com.example.initialproject.model.RequestListener;
import com.example.initialproject.model.RequestModelImpl;
import com.example.initialproject.view.MyView;

public class PresenterImpl implements Presenter, RequestListener {
    private RequestModelImpl requestModelImpl;
    private MyView myView;

    public PresenterImpl(MyView myView) {
        this.myView = myView;
        this.requestModelImpl = new RequestModelImpl();
    }

    @Override
    public void loadData(String url) {
        requestModelImpl.getData(PresenterImpl.this, url);
    }

    @Override
    public void onSuccess(String data) {
        myView.showData(data);
    }

    @Override
    public void onFail() {
        myView.showData("error");
    }
}

File: app/src/main/java/com/example/initialproject/presenter/PresenterImpl.kt
package com.example.initialproject.presenter

import com.example.initialproject.model.RequestListener
import com.example.initialproject.model.RequestModel
import com.example.initialproject.model.RequestModelImpl
import com.example.initialproject.view.MyView


class PresenterImpl(private val myView: MyView) : Presenter, RequestListener {
    private val requestModel: RequestModel = RequestModelImpl()

    override  fun loadData(url: String, type: Int) {
        requestModel.getData(this@PresenterImpl, url, type)
    }

    override fun onSuccess(data: String) {
        myView.showData(data)
    }

    override fun onFail() {
        myView.showData("error")
    }
}